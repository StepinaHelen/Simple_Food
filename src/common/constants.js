export const TYPES_LIST = {
  addItem: "ADD_ITEM",
  removeItem: "REMOVE_ITEM",
};

export const CATEGORIES = [
  "All",
  "Pizza",
  "Sushi",
  "Drinks",
  "Noodles",
  "Roles",
  "Salads",
];

export const ROUTES = {
  cartPage: "/cart",
  orderPage: "/order-page",
  orderHistoryPage: "/order-history",
};
