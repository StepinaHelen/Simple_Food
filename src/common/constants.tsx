import {
  ITYPES_LIST,
  IROUTES,
  IKEYQUERIES,
  IFORMS_FIELD_ITEM,
} from "./interfaces";

export const TYPES_LIST: ITYPES_LIST = {
  addItem: "ADD_ITEM",
  removeItem: "REMOVE_ITEM",
  clearCart: "CLEAR_CART",
};

export const CATEGORIES: string[] = [
  "All",
  "Pizza",
  "Sushi",
  "Drinks",
  "Noodles",
  "Salads",
];

export const ROUTES: IROUTES = {
  cartPage: "/cart",
  orderPage: "/order-page",
  orderHistoryPage: "/order-history",
};

export const KEYQUERIES: IKEYQUERIES = {
  orders: "orders",
  cards: "cards",
  orderHistoryPage: "/order-history",
};

export const FORMS_FIELD: IFORMS_FIELD_ITEM[] = [
  { id: 1, name: "name", title: "First Name" },
  { id: 2, name: "surName", title: "Last Name" },
  { id: 3, name: "phone", title: "Phone number" },
  { id: 4, name: "city", title: "City" },
  { id: 5, name: "street", title: "Street" },
];
