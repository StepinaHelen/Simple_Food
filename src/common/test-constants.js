export const cardItem = {
  id: 15,
  title: "Noodles with shrimp",
  price: 25,
  category: "noodles",
  amount: 1,
  img: "https://www.eatwell101.com/wp-content/uploads/2018/04/Shrimp-Zucchini-Noodles.jpg",
};

export const mockTitle = "Noodles with shrimp";

export const mockedData = {
  orderItems: [
    {
      id: 15,
      title: "Noodles with shrimp",
      price: 25,
      category: "noodles",
      amount: 1,
      img: "https://www.eatwell101.com/wp-content/uploads/2018/04/Shrimp-Zucchini-Noodles.jpg",
    },

    {
      id: 16,
      title: "Hawaiian pizza",
      price: 25,
      category: "pizza",
      amount: 1,
      img: "https://distfood.ru/upload/medialibrary/86a/86ab43892301045cb950f9874be29805.jpg",
    },
  ],
};

export const mockedLists = [
  {
    id: 15,
    title: "Noodles with shrimp",
    price: 25,
    category: "noodles",
    amount: 1,
    img: "https://www.eatwell101.com/wp-content/uploads/2018/04/Shrimp-Zucchini-Noodles.jpg",
  },
  {
    id: 16,
    title: "Hawaiian pizza",
    price: 25,
    category: "pizza",
    amount: 1,
    img: "https://distfood.ru/upload/medialibrary/86a/86ab43892301045cb950f9874be29805.jpg",
  },
];

export const mockedOrdersHistory = [
  {
    name: "Helen",
    surName: "Stepina",
    phone: "phone",
    city: "Wroclaw",
    street: "lotnicza",
    date: new Date(),
    items: mockedLists,
    totalAmount: 50,
  },
];
export const form = {
  name: "Helen",
  surName: "Stepina",
  phone: "phone",
  city: "Wroclaw",
  street: "lotnicza",
  date: new Date(),
};

export const cartContext = {
  items: mockedLists,
  totalAmount: 50,
};

export const newOrderToHistory = {
  name: "Olena",
  surName: "Stopina",
  phone: "0973601111",
  city: "Wroclaw",
  street: "Lotnicza",
  date: new Date(),
  items: mockedLists,
  totalAmount: 50,
};
