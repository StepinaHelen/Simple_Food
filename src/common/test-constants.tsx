import {
  ICartContextState,
  IForm,
  ICardItem,
  IOrdersHistoryItem,
  IOrders,
} from "./interfaces";

export const cardItem: ICardItem = {
  id: 15,
  title: "Noodles with shrimp",
  price: 25,
  category: "noodles",
  amount: 1,
  img: "https://www.eatwell101.com/wp-content/uploads/2018/04/Shrimp-Zucchini-Noodles.jpg",
};

export const mockTitle: string = "Noodles with shrimp";

export const mockedData: IOrders = {
  orderItems: [
    {
      id: 15,
      title: "Noodles with shrimp",
      price: 25,
      category: "noodles",
      amount: 1,
      img: "https://www.eatwell101.com/wp-content/uploads/2018/04/Shrimp-Zucchini-Noodles.jpg",
    },

    {
      id: 16,
      title: "Hawaiian pizza",
      price: 25,
      category: "pizza",
      amount: 1,
      img: "https://distfood.ru/upload/medialibrary/86a/86ab43892301045cb950f9874be29805.jpg",
    },
  ],
};

export const mockedLists: ICardItem[] = [
  {
    id: 15,
    title: "Noodles with shrimp",
    price: 25,
    category: "noodles",
    amount: 1,
    img: "https://www.eatwell101.com/wp-content/uploads/2018/04/Shrimp-Zucchini-Noodles.jpg",
  },
  {
    id: 16,
    title: "Hawaiian pizza",
    price: 25,
    category: "pizza",
    amount: 1,
    img: "https://distfood.ru/upload/medialibrary/86a/86ab43892301045cb950f9874be29805.jpg",
  },
];

export const mockedOrdersHistory: IOrdersHistoryItem[] = [
  {
    id: 1,
    name: "Helen",
    surName: "Stepina",
    phone: "phone",
    city: "Wroclaw",
    street: "lotnicza",
    date: "Wed Dec 21 2022 12:18:55 GMT+0100 (Central European Standard Time)",
    items: mockedLists,
    totalAmount: 50,
  },
];
export const form: IForm = {
  name: "Helen",
  surName: "Stepina",
  phone: "phone",
  city: "Wroclaw",
  street: "lotnicza",
  date: "22 Jan",
};

export const cartContext: ICartContextState = {
  items: mockedLists,
  totalAmount: 50,
};

export const newOrderToHistory = {
  name: "Olena",
  surName: "Stopina",
  phone: "0973601111",
  city: "Wroclaw",
  street: "Lotnicza",
  date: "Wed Dec 21 2022 12:18:55 GMT+0100 (Central European Standard Time)",
  items: mockedLists,
  totalAmount: 50,
};
