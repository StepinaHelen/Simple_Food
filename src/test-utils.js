import { render } from "@testing-library/react";
import { QueryClientProvider, QueryClient } from "react-query";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import { baseTheme } from "./styles/theme";
const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});
const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={baseTheme}>
      <QueryClientProvider contextSharing={true} client={queryClient}>
        <BrowserRouter>{children}</BrowserRouter>
      </QueryClientProvider>
    </ThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";

export { customRender as render };
