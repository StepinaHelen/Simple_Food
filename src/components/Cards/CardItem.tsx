import { useContext, useState } from "react";
import Button from "components/Button/Button";
import CartContext from "store/cart-context";
import { toast } from "react-toastify";
import {
  Item,
  WraperItem,
  AmountWrapper,
  BtnWrapper,
} from "components/Cards/CardsStyles";
import { ICardItem } from "common/interfaces";
import { useDispatch, useSelector } from "react-redux/es/exports";
import { cartActions } from "store/cart-slice";

const CardItem = (props: ICardItem) => {
  const dispatchFunction = useDispatch();

  // const cartContext = useContext(CartContext);
  const [value, setValue] = useState(1);
  const decrement = () => {
    if (value > 1) {
      setValue((previous) => previous - 1);
    }
  };

  const addToCartHandler = () => {
    dispatchFunction(
      cartActions.addItem({
        id: props.id,
        title: props.title,
        price: props.price,
        category: props.category,
        amount: value,
        img: props.img,
      })
    );
    // cartContext.addItem({
    //   id: props.id,
    //   title: props.title,
    //   price: props.price,
    //   category: props.category,
    //   amount: value,
    //   img: props.img,
    // });

    setValue(1);
    toast(`${props.title} added to cart successfully!`);
  };

  const increment = () => {
    setValue((previous) => previous + 1);
  };

  return (
    <Item>
      <img src={props.img} alt={props.title} />
      <WraperItem>
        <h2>{props.title}</h2>
        <p>
          Price: <span> {props.price}$</span>
        </p>
        <AmountWrapper>
          <p>Amount:</p>
          <button type="button" onClick={decrement} disabled={value === 0}>
            -
          </button>
          <span data-testid="value">{value}</span>
          <button type="button" onClick={increment}>
            +
          </button>
        </AmountWrapper>
        <BtnWrapper>
          <Button type="button" onClick={addToCartHandler}>
            Add
          </Button>
        </BtnWrapper>
      </WraperItem>
    </Item>
  );
};

export default CardItem;
