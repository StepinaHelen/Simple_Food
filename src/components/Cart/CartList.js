import CartItem from "./CartItem";
import CommonContainer from "../Base/Containers/CommonContainer";
import { useContext } from "react";
import CartContext from "../../store/cart-context";
import Button from "../Button/Button";
import { Link } from "react-router-dom";
import { List, TotalWrapper } from "./CartStyles";
import { ROUTES } from "../../common/constants";
import Notice from "../Notice/Notice";
import { useSelector } from "react-redux";

const CartList = () => {
  const cartContext = useContext(CartContext);

  const cartItems = useSelector((state) => {
    return state.cart.items;
  });

  const cartTotalAmount = useSelector((state) => {
    return state.cart.totalAmount;
  });

  return (
    <CommonContainer withMargin={true}>
      {cartItems && cartItems.length > 0 && (
        <>
          <List>
            {cartItems.map((el) => (
              <CartItem
                img={el.img}
                price={el.price}
                amount={el.amount}
                id={el.id}
                title={el.title}
                category={el.category}
                key={el.id}
              />
            ))}
          </List>
          <TotalWrapper>
            <p>Total:</p> <p>{cartTotalAmount}$</p>
          </TotalWrapper>
          <Link to={ROUTES.orderPage}>
            <Button className={"goToOrder"}>Go to order</Button>
          </Link>
        </>
      )}
      {!cartItems.length && (
        <Notice
          title="You don't have any items in your cart!"
          message="Please add some food to the cart. We're waiting for you :)"
        ></Notice>
      )}
    </CommonContainer>
  );
};

export default CartList;
