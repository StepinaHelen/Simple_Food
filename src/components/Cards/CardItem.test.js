import { render, screen } from "test-utils";
import userEvent from "@testing-library/user-event";
import CardItem from "./CardItem";

import CartContext from "store/cart-context";
import CartList from "components/Cart/CartList";
import { cardItem, mockTitle } from "common/test-constants";

describe("test CardItem", () => {
  it("CardItem renders", () => {
    render(<CardItem {...cardItem} />);
    expect(screen.getByText(mockTitle)).toBeInTheDocument();
    const img = screen.getByRole("img");
    expect(img.src).toContain(
      "https://www.eatwell101.com/wp-content/uploads/2018/04/Shrimp-Zucchini-Noodles.jpg"
    );
  });

  it("CardItem renders without data", () => {
    render(<CardItem />);
    expect(screen.queryByText(mockTitle)).toBeNull();
  });

  it("CardItem snapshot", () => {
    const view = render(<CardItem {...cardItem} />);
    expect(view).toMatchSnapshot();
  });

  it("CardItem snapshot with empty data", () => {
    const view = render(<CardItem />);
    expect(view).toMatchSnapshot();
  });

  it("click on the increment", async () => {
    render(<CardItem {...cardItem} />);
    const incrementBtn = screen.getByRole("button", { name: "+" });
    const value = await screen.findByTestId("value");
    expect(value.textContent).toContain("1");
    await userEvent.click(incrementBtn);
    expect(value.textContent).toContain("2");
  });

  it("click on the decrement", async () => {
    render(<CardItem {...cardItem} />);
    const incrementBtn = screen.getByRole("button", { name: "-" });
    const value = await screen.findByTestId("value");
    expect(value.textContent).toContain("1");
    userEvent.click(incrementBtn);
    expect(value.textContent).toContain("1");
  });

  it("click on the Add button", async () => {
    const addItem = jest.fn();
    render(
      <CartContext.Provider
        value={{
          addItem,
        }}
      >
        <CardItem {...cardItem} />
        <CartList />
      </CartContext.Provider>
    );
    userEvent.click(screen.getByText(/Add/));
    expect(addItem).toHaveBeenCalledTimes(1);
  });
});
