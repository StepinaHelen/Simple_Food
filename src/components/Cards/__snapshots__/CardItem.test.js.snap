// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test CardItem CardItem snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c4 {
  background-color: #6b068a;
  padding: 5px 27px;
  font-size: 16px;
  line-height: 19px;
  color: #ffffff;
  border-radius: 15px;
  font-size: 24px;
  line-height: 28px;
  border: none;
  outline: none;
  cursor: pointer;
}

.c4.login {
  padding: 9px 51px;
}

.c4:disabled {
  opacity: 0.7;
}

.c4:hover {
  background-color: #c913d6;
}

.c4.categories {
  margin-bottom: 12px;
  padding: 4px 12px;
  margin-right: 12px;
}

.c4.cardItem {
  font-size: 16px;
  line-height: 19px;
  padding: 5px 27px;
}

.c4.order {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4.goToOrder {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 13px 54px;
  font-weight: 600;
}

.c0 {
  margin: 15px;
  background-color: #fff;
  box-shadow: 0px 2px 1px rgb(0 0 0 / 20%),0px 1px 1px rgb(0 0 0 / 14%), 0px 1px 3px rgb(0 0 0 / 12%);
  border-radius: 0px 0px 4px 4px;
  margin-left: auto;
  margin-right: auto;
}

.c0 img {
  width: 100%;
  object-fit: contain;
}

.c1 {
  padding: 20px 20px;
}

.c1 h2 {
  font-weight: 600;
  font-size: 22px;
  line-height: 28px;
  margin-bottom: 20px;
}

.c1 p {
  margin-bottom: 20px;
  font-size: 20px;
  line-height: 23px;
  font-weight: 600;
}

.c1 p span {
  font-weight: 400;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 20px;
}

.c2 button {
  background-color: #6b068a;
  height: 20px;
  width: 20px;
  outline: none;
  border: none;
  border-radius: 5px;
  color: white;
}

.c2 p {
  margin: 0 12px 0 0;
}

.c2 span {
  margin-right: 3px;
  margin-left: 3px;
}

.c3 {
  text-align: right;
}

@media screen and (min-width:768px) {
  .c4.categories {
    padding: 8px 23px;
    margin-right: 24px;
  }
}

@media screen and (min-width:320px) {
  .c0 {
    width: 100%;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    width: calc((100% - 4 * 15px) / 2);
  }

  .c0 img {
    height: 226px;
  }
}

@media screen and (min-width:1200px) {
  .c0 {
    width: calc((100% - 8 * 15px) / 4);
  }

  .c0 img {
    height: 175px;
  }
}

<body>
    <div>
      <li
        class="c0"
      >
        <img
          alt="Noodles with shrimp"
          src="https://www.eatwell101.com/wp-content/uploads/2018/04/Shrimp-Zucchini-Noodles.jpg"
        />
        <div
          class="c1"
        >
          <h2>
            Noodles with shrimp
          </h2>
          <p>
            Price: 
            <span>
               
              25
              $
            </span>
          </p>
          <div
            class="c2"
          >
            <p>
              Amount:
            </p>
            <button
              type="button"
            >
              -
            </button>
            <span
              data-testid="value"
            >
              1
            </span>
            <button
              type="button"
            >
              +
            </button>
          </div>
          <div
            class="c3"
          >
            <button
              class="c4"
              type="button"
            >
              Add
            </button>
          </div>
        </div>
      </li>
    </div>
  </body>,
  "container": .c4 {
  background-color: #6b068a;
  padding: 5px 27px;
  font-size: 16px;
  line-height: 19px;
  color: #ffffff;
  border-radius: 15px;
  font-size: 24px;
  line-height: 28px;
  border: none;
  outline: none;
  cursor: pointer;
}

.c4.login {
  padding: 9px 51px;
}

.c4:disabled {
  opacity: 0.7;
}

.c4:hover {
  background-color: #c913d6;
}

.c4.categories {
  margin-bottom: 12px;
  padding: 4px 12px;
  margin-right: 12px;
}

.c4.cardItem {
  font-size: 16px;
  line-height: 19px;
  padding: 5px 27px;
}

.c4.order {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4.goToOrder {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 13px 54px;
  font-weight: 600;
}

.c0 {
  margin: 15px;
  background-color: #fff;
  box-shadow: 0px 2px 1px rgb(0 0 0 / 20%),0px 1px 1px rgb(0 0 0 / 14%), 0px 1px 3px rgb(0 0 0 / 12%);
  border-radius: 0px 0px 4px 4px;
  margin-left: auto;
  margin-right: auto;
}

.c0 img {
  width: 100%;
  object-fit: contain;
}

.c1 {
  padding: 20px 20px;
}

.c1 h2 {
  font-weight: 600;
  font-size: 22px;
  line-height: 28px;
  margin-bottom: 20px;
}

.c1 p {
  margin-bottom: 20px;
  font-size: 20px;
  line-height: 23px;
  font-weight: 600;
}

.c1 p span {
  font-weight: 400;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 20px;
}

.c2 button {
  background-color: #6b068a;
  height: 20px;
  width: 20px;
  outline: none;
  border: none;
  border-radius: 5px;
  color: white;
}

.c2 p {
  margin: 0 12px 0 0;
}

.c2 span {
  margin-right: 3px;
  margin-left: 3px;
}

.c3 {
  text-align: right;
}

@media screen and (min-width:768px) {
  .c4.categories {
    padding: 8px 23px;
    margin-right: 24px;
  }
}

@media screen and (min-width:320px) {
  .c0 {
    width: 100%;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    width: calc((100% - 4 * 15px) / 2);
  }

  .c0 img {
    height: 226px;
  }
}

@media screen and (min-width:1200px) {
  .c0 {
    width: calc((100% - 8 * 15px) / 4);
  }

  .c0 img {
    height: 175px;
  }
}

<div>
    <li
      class="c0"
    >
      <img
        alt="Noodles with shrimp"
        src="https://www.eatwell101.com/wp-content/uploads/2018/04/Shrimp-Zucchini-Noodles.jpg"
      />
      <div
        class="c1"
      >
        <h2>
          Noodles with shrimp
        </h2>
        <p>
          Price: 
          <span>
             
            25
            $
          </span>
        </p>
        <div
          class="c2"
        >
          <p>
            Amount:
          </p>
          <button
            type="button"
          >
            -
          </button>
          <span
            data-testid="value"
          >
            1
          </span>
          <button
            type="button"
          >
            +
          </button>
        </div>
        <div
          class="c3"
        >
          <button
            class="c4"
            type="button"
          >
            Add
          </button>
        </div>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test CardItem CardItem snapshot with empty data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c4 {
  background-color: #6b068a;
  padding: 5px 27px;
  font-size: 16px;
  line-height: 19px;
  color: #ffffff;
  border-radius: 15px;
  font-size: 24px;
  line-height: 28px;
  border: none;
  outline: none;
  cursor: pointer;
}

.c4.login {
  padding: 9px 51px;
}

.c4:disabled {
  opacity: 0.7;
}

.c4:hover {
  background-color: #c913d6;
}

.c4.categories {
  margin-bottom: 12px;
  padding: 4px 12px;
  margin-right: 12px;
}

.c4.cardItem {
  font-size: 16px;
  line-height: 19px;
  padding: 5px 27px;
}

.c4.order {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4.goToOrder {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 13px 54px;
  font-weight: 600;
}

.c0 {
  margin: 15px;
  background-color: #fff;
  box-shadow: 0px 2px 1px rgb(0 0 0 / 20%),0px 1px 1px rgb(0 0 0 / 14%), 0px 1px 3px rgb(0 0 0 / 12%);
  border-radius: 0px 0px 4px 4px;
  margin-left: auto;
  margin-right: auto;
}

.c0 img {
  width: 100%;
  object-fit: contain;
}

.c1 {
  padding: 20px 20px;
}

.c1 h2 {
  font-weight: 600;
  font-size: 22px;
  line-height: 28px;
  margin-bottom: 20px;
}

.c1 p {
  margin-bottom: 20px;
  font-size: 20px;
  line-height: 23px;
  font-weight: 600;
}

.c1 p span {
  font-weight: 400;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 20px;
}

.c2 button {
  background-color: #6b068a;
  height: 20px;
  width: 20px;
  outline: none;
  border: none;
  border-radius: 5px;
  color: white;
}

.c2 p {
  margin: 0 12px 0 0;
}

.c2 span {
  margin-right: 3px;
  margin-left: 3px;
}

.c3 {
  text-align: right;
}

@media screen and (min-width:768px) {
  .c4.categories {
    padding: 8px 23px;
    margin-right: 24px;
  }
}

@media screen and (min-width:320px) {
  .c0 {
    width: 100%;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    width: calc((100% - 4 * 15px) / 2);
  }

  .c0 img {
    height: 226px;
  }
}

@media screen and (min-width:1200px) {
  .c0 {
    width: calc((100% - 8 * 15px) / 4);
  }

  .c0 img {
    height: 175px;
  }
}

<body>
    <div>
      <li
        class="c0"
      >
        <img />
        <div
          class="c1"
        >
          <h2 />
          <p>
            Price: 
            <span>
               
              $
            </span>
          </p>
          <div
            class="c2"
          >
            <p>
              Amount:
            </p>
            <button
              type="button"
            >
              -
            </button>
            <span
              data-testid="value"
            >
              1
            </span>
            <button
              type="button"
            >
              +
            </button>
          </div>
          <div
            class="c3"
          >
            <button
              class="c4"
              type="button"
            >
              Add
            </button>
          </div>
        </div>
      </li>
    </div>
  </body>,
  "container": .c4 {
  background-color: #6b068a;
  padding: 5px 27px;
  font-size: 16px;
  line-height: 19px;
  color: #ffffff;
  border-radius: 15px;
  font-size: 24px;
  line-height: 28px;
  border: none;
  outline: none;
  cursor: pointer;
}

.c4.login {
  padding: 9px 51px;
}

.c4:disabled {
  opacity: 0.7;
}

.c4:hover {
  background-color: #c913d6;
}

.c4.categories {
  margin-bottom: 12px;
  padding: 4px 12px;
  margin-right: 12px;
}

.c4.cardItem {
  font-size: 16px;
  line-height: 19px;
  padding: 5px 27px;
}

.c4.order {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4.goToOrder {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 13px 54px;
  font-weight: 600;
}

.c0 {
  margin: 15px;
  background-color: #fff;
  box-shadow: 0px 2px 1px rgb(0 0 0 / 20%),0px 1px 1px rgb(0 0 0 / 14%), 0px 1px 3px rgb(0 0 0 / 12%);
  border-radius: 0px 0px 4px 4px;
  margin-left: auto;
  margin-right: auto;
}

.c0 img {
  width: 100%;
  object-fit: contain;
}

.c1 {
  padding: 20px 20px;
}

.c1 h2 {
  font-weight: 600;
  font-size: 22px;
  line-height: 28px;
  margin-bottom: 20px;
}

.c1 p {
  margin-bottom: 20px;
  font-size: 20px;
  line-height: 23px;
  font-weight: 600;
}

.c1 p span {
  font-weight: 400;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 20px;
}

.c2 button {
  background-color: #6b068a;
  height: 20px;
  width: 20px;
  outline: none;
  border: none;
  border-radius: 5px;
  color: white;
}

.c2 p {
  margin: 0 12px 0 0;
}

.c2 span {
  margin-right: 3px;
  margin-left: 3px;
}

.c3 {
  text-align: right;
}

@media screen and (min-width:768px) {
  .c4.categories {
    padding: 8px 23px;
    margin-right: 24px;
  }
}

@media screen and (min-width:320px) {
  .c0 {
    width: 100%;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    width: calc((100% - 4 * 15px) / 2);
  }

  .c0 img {
    height: 226px;
  }
}

@media screen and (min-width:1200px) {
  .c0 {
    width: calc((100% - 8 * 15px) / 4);
  }

  .c0 img {
    height: 175px;
  }
}

<div>
    <li
      class="c0"
    >
      <img />
      <div
        class="c1"
      >
        <h2 />
        <p>
          Price: 
          <span>
             
            $
          </span>
        </p>
        <div
          class="c2"
        >
          <p>
            Amount:
          </p>
          <button
            type="button"
          >
            -
          </button>
          <span
            data-testid="value"
          >
            1
          </span>
          <button
            type="button"
          >
            +
          </button>
        </div>
        <div
          class="c3"
        >
          <button
            class="c4"
            type="button"
          >
            Add
          </button>
        </div>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
